cli/commands/asset/outdated.go

```diff
@@ -0,0 +1,165 @@
+package asset
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+
+	goversion "github.com/hashicorp/go-version"
+	corev2 "github.com/sensu/sensu-go/api/core/v2"
+	"github.com/sensu/sensu-go/bonsai"
+	"github.com/sensu/sensu-go/cli"
+	"github.com/sensu/sensu-go/cli/client"
+	"github.com/sensu/sensu-go/cli/commands/flags"
+	"github.com/sensu/sensu-go/cli/commands/helpers"
+	"github.com/sensu/sensu-go/cli/elements/table"
+	"github.com/sensu/sensu-go/types"
+	"github.com/spf13/cobra"
+)
+
+// OutdatedCommand adds a command that allows users to list outdated assets
+// that have been added from Bonsai.
+func OutdatedCommand(cli *cli.SensuCli) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "outdated",
+		Short: "lists any assets installed from Bonsai that have newer versions available",
+		RunE:  outdatedCommandExecute(cli),
+	}
+
+	helpers.AddFormatFlag(cmd.Flags())
+	helpers.AddAllNamespace(cmd.Flags())
```

Comment:
Are these flags all necessary here for this command? `outdated` only refers to bonsai, not sensu resources right? So the namespace, field selector, label selector, and chunk size flags might not be needed.

Reply:
Outdated still has to look up assets from the Sensu API in order to find out what assets are installed and which of them are Bonsai assets. I'm unsure on what exactly the field, label & chunk size flags do so we may not need them. This was copied over from the `asset list` command.

Reply:
Conceptually, the flags for {field,label} selectors allows `sensuctl` to ask the backend to only return resources that match certain predicates and the chunk size flag is to make `sensuctl` use the API's pagination facilities to build a resource list chunk by chunk (pages) rather than all in one go.

So you are right, we don't need them here!

(technically we could make use of `--chunk-size` if we wanted to, to fetch the list of current assets from the API, but that would only be useful if one had 1000s of assets and a crappy connection. Let's not bother.)

Reply:
Agreed - this & the `--all-namespaces` flags can be added in a future release.